libdispatch/CMakeLists.txt

```diff
@@ -4,57 +4,62 @@
 daux.c dinstance.c dinstance_intern.c
 dcrc32.c dcrc32.h dcrc64.c ncexhash.c ncxcache.c ncjson.c ds3util.c dparallel.c dmissing.c)
 
 # Netcdf-4 only functions. Must be defined even if not used
-SET(libdispatch_SOURCES ${libdispatch_SOURCES} dgroup.c dvlen.c dcompound.c dtype.c denum.c dopaque.c dfilter.c)
+set(libdispatch_SOURCES ${libdispatch_SOURCES} dgroup.c dvlen.c dcompound.c dtype.c denum.c dopaque.c dfilter.c)
 
-IF(BUILD_V2)
-  SET(libdispatch_SOURCES ${libdispatch_SOURCES} dv2i.c)
-ENDIF(BUILD_V2)
+if(BUILD_V2)
+  set(libdispatch_SOURCES ${libdispatch_SOURCES} dv2i.c)
+endif(BUILD_V2)
 
-IF(ENABLE_BYTERANGE)
-  SET(libdispatch_SOURCES ${libdispatch_SOURCES} dhttp.c)
+if(ENABLE_BYTERANGE)
+  set(libdispatch_SOURCES ${libdispatch_SOURCES} dhttp.c)
 ENDIF(ENABLE_BYTERANGE)
 
 IF(ENABLE_S3)
-  IF(ENABLE_S3_INTERNAL)
-    SET(libdispatch_SOURCES ${libdispatch_SOURCES} ncs3sdk_h5.c nch5s3comms.c nch5s3comms.h nccurl_sha256.c nccurl_sha256.h nccurl_hmac.c nccurl_hmac.h nccurl_setup.h)
-  ELSE()
-    SET(libdispatch_SOURCES ${libdispatch_SOURCES} ncs3sdk_aws.cpp awsincludes.h)
-  ENDIF()
-ENDIF()
+  if(ENABLE_S3_INTERNAL)
+    set(libdispatch_SOURCES ${libdispatch_SOURCES} ncs3sdk_h5.c nch5s3comms.c nch5s3comms.h nccurl_sha256.c nccurl_sha256.h nccurl_hmac.c nccurl_hmac.h nccurl_setup.h)
+  else()
+    set(libdispatch_SOURCES ${libdispatch_SOURCES} ncs3sdk_aws.cpp awsincludes.h)
+  endif()
+endif()
 
-IF(REGEDIT)
-  SET(libdispatch_SOURCES ${libdispatch_SOURCES} dreg.c)
-ENDIF(REGEDIT)
+if(REGEDIT)
+  set(libdispatch_SOURCES ${libdispatch_SOURCES} dreg.c)
+endif(REGEDIT)
 
 add_library(dispatch OBJECT ${libdispatch_SOURCES})
-IF(STATUS_PARALLEL)
+
+if (USE_HDF5)
+  target_include_directories(dispatch PRIVATE ${HDF5_INCLUDE_DIR})
+endif(USE_HDF5)
+
+if(STATUS_PARALLEL)
     target_link_libraries(dispatch PUBLIC MPI::MPI_C)
-ENDIF(STATUS_PARALLEL)
+endif(STATUS_PARALLEL)
 
-IF(ENABLE_NCZARR)
+if(ENABLE_NCZARR)
     target_include_directories(dispatch PUBLIC ../libnczarr)
-ENDIF(ENABLE_NCZARR)
+endif(ENABLE_NCZARR)
 
-IF(ENABLE_S3)
-  IF(ENABLE_S3_AWS)
+if(ENABLE_S3)
+  if(ENABLE_S3_AWS)
     target_include_directories(dispatch PUBLIC ${AWSSDK_INCLUDE_DIRS})
-    IF(NOT MSVC)
+    if(NOT MSVC)
       target_compile_features(dispatch PUBLIC cxx_std_11)
-    ENDIF()
-  ELSE()
+    endif()
+  else()
     target_include_directories(dispatch PUBLIC ../libncxml)
-  ENDIF()
-ENDIF()
+  endif()
+endif()
 
-IF(ENABLE_TESTS)
+if(ENABLE_TESTS)
     BUILD_BIN_TEST(ncrandom)
-ENDIF()
+endif()
 
-FILE(GLOB CUR_EXTRA_DIST RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_CURRENT_SOURCE_DIR}/*.h ${CMAKE_CURRENT_SOURCE_DIR}/*.c)
-SET(CUR_EXTRA_DIST ${CUR_EXTRA_DIST} CMakeLists.txt Makefile.am)
+file(GLOB CUR_EXTRA_DIST RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_CURRENT_SOURCE_DIR}/*.h ${CMAKE_CURRENT_SOURCE_DIR}/*.c)
```

Comment:
Don't use glob, it affects reconfiguration.

Reply:
this is for another PR